File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0;
        int s = flowerbed.size();
        for(int i = 0 ; i < s ; i++){
            if(flowerbed[i] == 0) {

            
            bool left = (i == 0 || flowerbed[i-1] == 0);
            bool right = ( i == s-1 || flowerbed[i+1] == 0);
            if(left && right){
                flowerbed[i] = 1;
                count++;
            }
            }
        }
        return n <= count;
        
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int count = 0;
        int s = flowerbed.length;

        for (int i = 0; i < s; i++) {
            if (flowerbed[i] == 0) {
                // Check if the left and right plots are empty or out of bounds
                boolean emptyLeft = (i == 0 || flowerbed[i - 1] == 0);
                boolean emptyRight = (i == s - 1 || flowerbed[i + 1] == 0);

                if (emptyLeft && emptyRight) {
                    flowerbed[i] = 1;
                    count++;
                }
            }
        }

        return count >= n;
    }
}